src/enable_variable_ops/ngraph_encapsulate_op.cc

```diff
@@ -172,22 +173,17 @@ class NGraphEncapsulateOp : public OpKernel {
       // m_freshness_tracker->Unref();
     }
 
-    string node_name = "_ngraph_cluster_" + to_string(m_ngraph_cluster);
-    // TODO(malikshr) : Could be erroreneous we dont know if this
-    // destructor is called at the very end, if some modifier that uses this
-    // tensor is still active.
     for (int i = 0; i < m_number_outputs; i++) {
-      string key = NGraphCatalog::CreateNodeKey(m_graph_id, node_name, i);
+      string key = NGraphCatalog::CreateNodeKey(m_graph_id, name(), i);
       if (NGraphCatalog::ExistsInEncapOutputTensorMap(key)) {
-        auto temp = NGraphCatalog::GetTensorFromEncapOutputTensorMap(key);
-        temp.reset();
         NGraphCatalog::DeleteFromEncapOutputTensorMap(key);
+        NGRAPH_VLOG(2) << "Deleting from output tensor map " << key;
```

Comment:
Checking for Exists before deleting though it is not required to log which tensors are being deleted.